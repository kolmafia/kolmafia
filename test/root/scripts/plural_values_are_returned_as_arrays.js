const { printHtml } = require("kolmafia");

[
  'Ed the Undying',
  'warwelf',
  'lihc',
  'ancient protector spirit'
].forEach((name) => {
  const monster = Monster.get(name);

  printHtml(JSON.stringify({
    name,
    attackElement: (monster.attackElement || '').toString(),
    attackElements: monster.attackElements.map((element) => element.toString()),
    images: monster.images,
    randomModifiers: monster.randomModifiers,
    subTypes: monster.subTypes,
  }, null, 2));
});

[
  'Beaten up',
  'Tomato Power'
].forEach((name) => {
  const effect = Effect.get(name);

  printHtml(JSON.stringify({
    name,
    all: effect.all,
  }, null, 2));
});