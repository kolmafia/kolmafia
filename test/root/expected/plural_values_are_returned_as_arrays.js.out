{
  "name": "Ed the Undying",
  "attackElement": "none",
  "attackElements": [
    "none"
  ],
  "images": [
    "ed.gif",
    "ed2.gif",
    "ed3.gif",
    "ed4.gif",
    "ed5.gif",
    "ed6.gif",
    "ed7.gif"
  ],
  "randomModifiers": [],
  "subTypes": []
}
{
  "name": "warwelf",
  "attackElement": "none",
  "attackElements": [
    "none"
  ],
  "images": [
    "werewolf.gif"
  ],
  "randomModifiers": [],
  "subTypes": []
}
{
  "name": "lihc",
  "attackElement": "spooky",
  "attackElements": [
    "sleaze",
    "spooky"
  ],
  "images": [
    "lich.gif"
  ],
  "randomModifiers": [],
  "subTypes": []
}
{
  "name": "ancient protector spirit",
  "attackElement": "spooky",
  "attackElements": [
    "spooky"
  ],
  "images": [
    "protspirit.gif"
  ],
  "randomModifiers": [],
  "subTypes": [
    "ghost"
  ]
}
{
  "name": "Beaten up",
  "all": [
    "# losing in combat"
  ]
}
{
  "name": "Tomato Power",
  "all": [
    "use 1 tomato juice of powerful power",
    "use 1 evil tomato juice of powerful power"
  ]
}